api/src/pcapi/core/token/__init__.py

```diff
@@ -94,9 +106,44 @@ def create(cls, type_: TokenType, ttl: timedelta | None, user_id: int, data: dic
         token._log(cls._TokenAction.CREATE)
         return token
 
+
+@dataclasses.dataclass(frozen=True)
+class SixDigitsToken(AbstractToken):
     @classmethod
-    def token_exists(cls, type_: TokenType, user_id: int) -> bool:
-        return app.redis_client.exists(cls._get_redis_key(type_, user_id))  # type: ignore [attr-defined]
+    def _get_redis_extra_data_key(cls, type_: TokenType, user_id: int) -> str:
+        return f"pcapi:token:data:{type_.value}_{user_id}"
 
-    def _log(self, action: _TokenAction) -> None:
-        logger.info("[TOKEN](%s){%i, %s, %s}", action.value, self.user_id, self.type_.value, self.encoded_token)
+    @classmethod
+    def load_without_checking(
+        cls,
+        encoded_token: str,
+        type_: TokenType,
+        user_id: int,
+    ) -> "SixDigitsToken":
+        try:
+            data = json.loads(app.redis_client.get(cls._get_redis_extra_data_key(type_=type_, user_id=user_id)))  # type: ignore [attr-defined]
+        except:
+            raise InvalidToken()
+        return cls(type_, user_id, encoded_token, data)
+
+    @classmethod
+    def load_and_check(cls, encoded_token: str, type_: TokenType, user_id: int) -> "SixDigitsToken":
+        token = cls.load_without_checking(encoded_token, type_=type_, user_id=user_id)
+        token.check(type_, user_id)
+        return token
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏Les 2 méthodes `load_and_check` font sensiblement la même chose. Elle peut rester dans la classe parente